File: app/miscellaneous/file/FileHandler.java
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package miscellaneous.file;

import java.io.File;
import java.io.IOException;
import java.nio.file.Path;
import models.attachment.Attachment;
import models.attachment.AttachmentContainer;
import play.libs.Files;
import play.mvc.Http;

public interface FileHandler {
    byte[] read(File file);
    String read(String path);
    String encodeAndDelete(File file);
    String getContentDisposition(File file);
    String createFilePath(String... pathParams);
    String getAttachmentPath();
    void removeAttachmentFile(String filePath);
    void removePrevious(AttachmentContainer container);
    Attachment createNew(Http.MultipartFormData.FilePart<?> file, String path);
    Attachment createNew(String fileName, String contentType, String path);
    void copyFile(Files.TemporaryFile sourceFile, File destFile) throws IOException;
    void copyFile(Path sourceFile, File destFile) throws IOException;
}

File: app/services/file/FileHandler.scala
// SPDX-FileCopyrightText: 2024 The members of the EXAM Consortium
//
// SPDX-License-Identifier: EUPL-1.2

package services.file

import models.attachment.{Attachment, AttachmentContainer}
import play.api.libs.Files.TemporaryFile

import java.io.File
import java.nio.file.Path

trait FileHandler:
  def read(file: File): Array[Byte]
  def read(path: String): String
  def encodeAndDelete(file: File): String
  def getContentDisposition(file: File): String
  @annotation.varargs
  def createFilePath(pathParams: String*): String
  def getAttachmentPath: String
  def removeAttachmentFile(filePath: String): Unit
  def removePrevious(container: AttachmentContainer): Unit
  def createNew(fileName: String, contentType: String, path: String): Attachment
  def copyFile(sourceFile: TemporaryFile, destFile: File): Unit
  def copyFile(sourceFile: Path, destFile: File): Unit
